source/_docs/automation/templating.markdown

```diff
@@ -13,18 +13,20 @@ The variable `this` is the [state object](/docs/configuration/state_object) of t
 
 ## Available Trigger Data
 
-The following tables show the available trigger data per platform.
+The variable `trigger` is an object that contains details about which trigger activated the automation.  The data can be used by templates to modify the actions that are performed by the automation or displayed in a message.  For example, you could create an automation that that can be triggered by multiple sensors and then use the location of the sensor to specify a light to activate; or you could send a notification that contains the friendly name of the sensor that triggered it.
+
+Each [trigger platform](/docs/automation/trigger/#event-trigger) can include additional data specific to that platform.
 
 ### All
 
-The following describes trigger data associated with all platforms.
+Triggers from all platform will include the following data.
 
 | Template variable | Data |
 | ---- | ---- |
-| `trigger.id` | Optional trigger `id`, or index of the trigger.
+| `trigger.id` | The [`id` of the trigger](/docs/automation/trigger/#trigger-id).  If `id` is omitted, this will be set to the trigger index.
 | `trigger.idx` | Index of the trigger. (The first trigger idx is `0`.)
 
-### Calendar
+### [Calendar](/docs/automation/trigger/#calendar-trigger)
```

Comment:
I think adding links to different pages on headings is actually not a good idea.

Instead, we should link to the paragraph itself.

This comment applies to every occurrence below.

Reply:
@frenck - Can you clarify what you mean by "link to the paragraph"?  Do you mean something similar to what I suggested in the PR description?

> I don't know if there are guidelines around whether or not to include links in headers, but for this list it seemed like the best option, but I'm happy to move the links into sentences right underneath the headers (e.g. "This is the data available in the [Calendar Trigger](/docs/automation/trigger/#calendar-trigger)").

If so, I can add a line of text at the start of each table that includes a link like that.

Reply:
Most of these section ONLY contain a table, so there was no obvious place to add a link without adding extra text.

Reply:
> If so, I can add a line of text at the start of each table that includes a link like that.

Yeah or a simple text/link to each paragraph with "Read more about configuring this trigger here".

The problem is that more often, headers in documentation (in general) are anchor links, in this case, it links to a different piece of documentation; which I think is unexpected/not good behavior.

Reply:
Ah yes, I've been using the header anchors a BUNCH for this PR!  That'd definitely be weird to have both together.  

I'll add the links inside the sections this afternoon.